vortex-layout/src/layouts/chunked/scan.rs

```diff
@@ -29,13 +31,19 @@ use crate::{LayoutData, LayoutEncoding, RowMask};
 /// boundary we will read the chunk twice). However, if we have overlapping row ranges, as can
 /// happen if the parent is performing multiple scans (filter + projection), then we may read the
 /// same chunk many times.
+#[derive(Clone, Debug)]
 pub struct ChunkedScan {
     layout: LayoutData,
     scan: Scan,
     dtype: DType,
     ctx: ContextRef,
+    // Shared stats table scanner
     stats_scanner: Arc<RwLock<Box<dyn Scanner>>>,
```

Comment:
In a FLUP I'm going to turn `Box<dyn LayoutScan>` into `Arc<dyn LayoutScan>`, and then `create_scanner(self: Arc<Self>, ...)`. 

Since it's a common pattern to share internally mutable state amongst scanners and this way we won't need all these internal arcs.